core/src/main/scala/org/apache/spark/deploy/security/HiveDelegationTokenProvider.scala

```diff
@@ -92,8 +93,8 @@ private[security] class HiveDelegationTokenProvider
         s"$principal at $metastoreUri")
 
       doAsRealUser {
-        val hive = Hive.get(conf, classOf[HiveConf])
-        val tokenStr = hive.getDelegationToken(currentUser.getUserName(), principal)
+        metaStoreClient = new HiveMetaStoreClient(conf.asInstanceOf[HiveConf])
+        val tokenStr = metaStoreClient.getDelegationToken(currentUser.getUserName, principal)
```

Comment:
As you mentioned in PR description, you tested this in a kerberized cluster.
At that time, did you test this PR against both HMS 1.x and 2.x?

Reply:
Yes, both HMS 1.x and 2.x